File: web/plugins/decimal_part.js
import { WebPlugin } from "../engine/plugin.js";

export class DecimalPartPlugin extends WebPlugin {
    constructor() {
        super("小数部分");
    }

    async execute(line, context) {
        const expression = line.replace(/^小数部分\s*/, "");

        const value = Number(
            context.parser.parseValue(
                expression,
                context.runtime
            )
        );

        const integerPart = Math.trunc(value);
        const decimalPart = value - integerPart;

        const roundedDecimalPart =
            Math.round(
                (decimalPart + Number.EPSILON) * 1e12
            ) / 1e12;

        context.runtime.write(
            roundedDecimalPart
        );
    }
}

File: pc/plugins/decimal_part.py
import re
from core.plugin import Plugin


class DecimalPartPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'小数部分\s+(.+)', line)

        if not match:
            return None

        value = match.group(1)

        return (
            f"print(round("
            f"({value}) - int({value}), "
            f"12"
            f"))"
        )